# STATISTICS EXERCISE - SOLUTION

## __Import Library__

In [42]:
import pandas as pd
pd.set_option('display.max_columns', None)

## __Open Dataset__

In [5]:
df = pd.read_csv('/Users/dymaslawula/Documents/data001/melb_data.csv')
df.head()

Suburb           Address  Rooms Type      Price Method SellerG  \
0  Abbotsford      85 Turner St      2    h  1480000.0      S  Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000.0      S  Biggin   
2  Abbotsford      5 Charles St      3    h  1465000.0     SP  Biggin   
3  Abbotsford  40 Federation La      3    h   850000.0     PI  Biggin   
4  Abbotsford       55a Park St      4    h  1600000.0     VB  Nelson   

        Date  Distance  Postcode  Bedroom2  Bathroom  Car  Landsize  \
0  3/12/2016       2.5    3067.0       2.0       1.0  1.0     202.0   
1  4/02/2016       2.5    3067.0       2.0       1.0  0.0     156.0   
2  4/03/2017       2.5    3067.0       3.0       2.0  0.0     134.0   
3  4/03/2017       2.5    3067.0       3.0       2.0  1.0      94.0   
4  4/06/2016       2.5    3067.0       3.0       1.0  2.0     120.0   

   BuildingArea  YearBuilt CouncilArea  Lattitude  Longtitude  \
0           NaN        NaN       Yarra   -37.7996    144.9984   
1          79.0     1900.0       Yarra   -37.8079    144.9934   
2         150.0     1900.0       Yarra   -37.8093    144.9944   
3           NaN        NaN       Yarra   -37.7969    144.9969   
4         142.0     2014.0       Yarra   -37.8072    144.9941   

              Regionname  Propertycount  
0  Northern Metropolitan         4019.0  
1  Northern Metropolitan         4019.0  
2  Northern Metropolitan         4019.0  
3  Northern Metropolitan         4019.0  
4  Northern Metropolitan         4019.0

<hr>

## **No. 1**


Show Descriptive Statistics on **Rooms, Distance, Landsize**, and **Price** Variables!

In [6]:
df[['Rooms', 'Distance','Landsize','Price']].describe()

Rooms      Distance       Landsize         Price
count  13580.000000  13580.000000   13580.000000  1.358000e+04
mean       2.937997     10.137776     558.416127  1.075684e+06
std        0.955748      5.868725    3990.669241  6.393107e+05
min        1.000000      0.000000       0.000000  8.500000e+04
25%        2.000000      6.100000     177.000000  6.500000e+05
50%        3.000000      9.200000     440.000000  9.030000e+05
75%        3.000000     13.000000     651.000000  1.330000e+06
max       10.000000     48.100000  433014.000000  9.000000e+06

<hr>

## **No. 2**

a) Berapa jumlah **Rooms** pada rumah yang harganya paling mahal?

In [8]:
df[df['Price']==df['Price'].max()][['Rooms']]

Rooms
12094      3

b) Berapa rata-rata harga rumah yang jarak ke CBD (variabel **Distance**) terdekat?

In [10]:
df[df['Distance']==df['Distance'].min()]['Price'].mean()

671166.6666666666

c) Berapa rata-rata harga rumah yang luas tanahnya (variabel **Landsize**) 1 kali standar deviasi di atas rata-rata luas tanah?

In [12]:
land_mean = df['Landsize'].mean()
land_std = df['Landsize'].std()

df[df['Landsize']> (land_mean + land_std)]['Price'].mean()

880776.6666666666

d) Berapa rentang (_range_) harga rumah (variabel **Price**) pada region ``Southern Metropolitan ``?

In [14]:
df_sm = df[df['Regionname']=='Southern Metropolitan']

price_range = df_sm['Price'].max() - df_sm['Price'].min()

price_range

7869000.0

<hr>

## **No. 3**

 * Outlier values are values that are below `` Q1 - (1.5 * IQR) or above Q3 + (1.5 * IQR)``
 * First calculate Q1, Q2, and IQR in the **Price** column (House Price)
 * Calculate how many outlier values are in the **Price** column

In [16]:
import numpy as np

df['Price'] = df['Price'].apply(np.int64)

In [17]:
q1 = df['Price'].describe()['25%']
q3 = df['Price'].describe()['75%']


iqr = q3 - q1
iqr

680000.0

In [18]:
batas_bawah = int(q1 - (1.5*iqr))
batas_atas = int(q3 + (1.5*iqr))

In [20]:
outlier_price = df[(df['Price'] < batas_bawah) | (df['Price'] > batas_atas)][['Price']]
outlier_price.head()

Price
80   2850000
92   2615000
97   2575000
102  3010000
103  2800000

In [21]:
# Number of outliers in column 'Price'

len(outlier_price)

612

In [25]:
outlier_price.index

Int64Index([   80,    92,    97,   102,   103,   108,   112,   118,   124,
              233,
            ...
            13466, 13468, 13474, 13483, 13518, 13521, 13523, 13553, 13555,
            13578],
           dtype='int64', length=612)

<hr>

## **No. 4**

#### **``Shapiro Wilk Test``**

Testing the assumption of normality of the Variable Price ``Price``

#### **Interpretation**
* If the P-Value of the Shapiro Wilk Test is larger than 0.05, we assume a normal distribution
* If the P-Value of the Shapiro Wilk Test is smaller than 0.05, we do not assume a normal distributio

In [26]:
from scipy.stats import shapiro

sw_statistic, sw_pvalue = shapiro (df['Price'])

if sw_pvalue > 0.05:
    print('P-Value: {}. So, We assume a normal distribution'.format(sw_pvalue))
else:
    print('P-Value: {}. So, We We do not assume a normal distribution'.format(sw_pvalue))

P-Value: 0.0. So, We We do not assume a normal distribution


/Users/dymaslawula/opt/anaconda3/lib/python3.9/site-packages/scipy/stats/morestats.py:1760: UserWarning: p-value may not be accurate for N > 5000.
  warnings.warn("p-value may not be accurate for N > 5000.")


#### **``D’Agostino and Pearson’s Test``**

Testing the assumption of normality of the Variable ``Price``

#### **Interpretation**
* If the P-Value of the D'Agostino-Pearson Test is larger than 0.05, we assume a normal distribution
* If the P-Value of the D'Agostino-Pearson Test is smaller than 0.05, we do not assume a normal distribution

In [30]:
from scipy.stats import normaltest
dp_statistic, dp_value = normaltest(df['Price'])

if dp_value > 0.05:
    print('P-Value: {}. So, We assume a normal distribution'.format(dp_value))
else:
    print('P-Value: {}. So, We We do not assume a normal distribution'.format(dp_value))

P-Value: 0.0. So, We We do not assume a normal distribution


<hr>

## **No. 5**

If it is assumed to have met the parametric statistical requirements, calculate how strong the correlation between the **Rooms** variable and **Price** is, and the direction of the correlation!!

In [31]:
df['Rooms'].corr(df['Price'], method = 'pearson')

0.49663367618653

<hr>

## **No. 6**

if it is assumed to have met the parametric statistical requirements, which of the several **Regionnames** with the number of rooms (**Rooms**) has the strongest correlation with the house price variable (**Price**)!

In [33]:
df['Regionname'].unique()

array(['Northern Metropolitan', 'Western Metropolitan',
       'Southern Metropolitan', 'Eastern Metropolitan',
       'South-Eastern Metropolitan', 'Eastern Victoria',
       'Northern Victoria', 'Western Victoria'], dtype=object)

In [34]:
for i in df['Regionname'].unique():
    print('{} = '.format(i) + str(df[df['Regionname'] == i]['Rooms'].corr(df[df['Regionname']==i]['Price'], method='pearson')))

Northern Metropolitan = 0.3928134583022406
Western Metropolitan = 0.476030293843498
Southern Metropolitan = 0.6837438462796438
Eastern Metropolitan = 0.5029928557468843
South-Eastern Metropolitan = 0.26779944081140417
Eastern Victoria = 0.3144233953469533
Northern Victoria = 0.3611442989081273
Western Victoria = 0.5518865952271457


In [36]:
corr = []
for i in df['Regionname'].unique():
    corr.append(df[df['Regionname'] == i]['Rooms'].corr(df[df['Regionname']==i]['Price'], method='pearson'))
    
corr

[0.3928134583022406,
 0.476030293843498,
 0.6837438462796438,
 0.5029928557468843,
 0.26779944081140417,
 0.3144233953469533,
 0.3611442989081273,
 0.5518865952271457]

In [43]:
pd.DataFrame(data = corr, index = df['Regionname'].unique(), columns = ['Correlation Price vs Rooms']
            ).sort_values('Correlation Price vs Rooms', ascending = False)

Correlation Price vs Rooms
Southern Metropolitan                         0.683744
Western Victoria                              0.551887
Eastern Metropolitan                          0.502993
Western Metropolitan                          0.476030
Northern Metropolitan                         0.392813
Northern Victoria                             0.361144
Eastern Victoria                              0.314423
South-Eastern Metropolitan                    0.267799

<hr>

## **Nomor 7**

If it is assumed that it **does not** meet the parametric statistical requirements, calculate how strong the correlation between the **Distance** variable and **Price** is, and the direction of the correlation?

In [44]:
df['Distance'].corr(df['Price'], method='spearman')

-0.12998984131880323

<hr>

## **No. 8**

If it is assumed that the parametric statistical requirements have been met, which of the several Regionnames **Regionname** the variable distance to the CBD (**Distance**) has the least correlation with the house price variable (**Price**)! And also display the most correlated Regionname of these two variables!

In [46]:
for i in df['Regionname'].unique():
    print(f'{i} = ' + str(df[df['Regionname'] == i]['Distance'].corr(df[df['Regionname']==i]['Price'], method= 'pearson')))

Northern Metropolitan = -0.3685519961790508
Western Metropolitan = -0.3340855504709213
Southern Metropolitan = 0.002489149218933446
Eastern Metropolitan = -0.1243556285268369
South-Eastern Metropolitan = -0.28556405189080514
Eastern Victoria = -0.34569916122464595
Northern Victoria = 0.08956649327360575
Western Victoria = -0.04606415345276123


In [47]:
corr = []
for i in df['Regionname'].unique():
    corr.append(df[df['Regionname'] == i]['Distance'].corr(df[df['Regionname']==i]['Price'], method= 'pearson'))
    
pd.DataFrame(data = corr, index = df['Regionname'].unique(), columns = ['Correlation Price vs Distance']
            ).sort_values('Correlation Price vs Distance', ascending=True)

Correlation Price vs Distance
Northern Metropolitan                           -0.368552
Eastern Victoria                                -0.345699
Western Metropolitan                            -0.334086
South-Eastern Metropolitan                      -0.285564
Eastern Metropolitan                            -0.124356
Western Victoria                                -0.046064
Southern Metropolitan                            0.002489
Northern Victoria                                0.089566

* Regionname with the least correlation or the weakest correlation in the two variables above is **Sourthern Metropolitan** (0.002489 or 0.24 percent).
* Regionname most correlated in the two variables above is **Northern Metropolitan** (-0.368552 or - 36.86 percent)

<hr>

## **No. 9**

Test the hypothesis whether the purchase method/**Method** is related to **Type** / house type (dependent) or not related (independent)!

In [48]:
df_ct = pd.crosstab(df['Type'], df['Method'])
df_ct

Method    PI     S  SA    SP   VB
Type                             
h       1069  6507  66  1079  728
t        134   723   7   143  107
u        361  1792  19   481  364

#### **Hypothesis:**
* Ho : The two variables are independent
* Ha :  The two variables are dependent.

In [51]:
from scipy.stats import chi2_contingency

stat, p, dof, expected = chi2_contingency(df_ct)

# conclusion
alpha = 0.05
print('P value:', p)

if p <= alpha:
    print("Dependent (Reject H0)")
else:
    print('Independent (Accept H0)')

P value: 3.183702485154384e-22
Dependent (Reject H0)


<hr>

## **No. 10**

Test the hypothesis whether the average house price (**Price**) between Regionnames is different or the same!

It is assumed that the parametric statistical requirements are met

**Hipotesis**:
* Ho : μ1 = μ2 = … = μk (The average house price between **Regionname** is the same)  
* Ha : There is at least one pair of **Regionname** whose average house prices are not the same, but we don't know which one

In [52]:
df['Regionname'].unique()

array(['Northern Metropolitan', 'Western Metropolitan',
       'Southern Metropolitan', 'Eastern Metropolitan',
       'South-Eastern Metropolitan', 'Eastern Victoria',
       'Northern Victoria', 'Western Victoria'], dtype=object)

In [53]:
from scipy.stats import f_oneway

f_oneway(df[df['Regionname'] == 'Northern Metropolitan']['Price'],
         df[df['Regionname'] == 'Western Metropolitan']['Price'],
         df[df['Regionname'] == 'Southern Metropolitan']['Price'],
         df[df['Regionname'] == 'Eastern Metropolitan']['Price'],
         df[df['Regionname'] == 'South-Eastern Metropolitan']['Price'],
         df[df['Regionname'] == 'Eastern Victoria']['Price'],
         df[df['Regionname'] == 'Northern Victoria']['Price'],
         df[df['Regionname'] == 'Western Victoria']['Price'],
        )

F_onewayResult(statistic=284.40018651008097, pvalue=0.0)

If the p-value is less than equal to 0.05 then there is sufficient evidence to reject Ho.
It means that Ha is accepted by a Regionname couple whose average house price difference is significant, but they don't know which one.

In [54]:
values = []
for i in df['Regionname'].unique():
    values.append(round(df[df['Regionname']==i]['Price'].mean(), 2))
    
pd.DataFrame(data=values, index=df['Regionname'].unique(), columns=['Price Mean']
            ).sort_values(by='Price Mean', ascending=False)

Price Mean
Southern Metropolitan       1372963.37
Eastern Metropolitan        1104079.63
South-Eastern Metropolitan   922943.78
Northern Metropolitan        898171.08
Western Metropolitan         866420.52
Eastern Victoria             699980.79
Northern Victoria            594829.27
Western Victoria             397523.44